src/render.rs

```diff
@@ -53,15 +54,23 @@ pub struct Renderer<'a> {
     context: Value,
     tera: &'a Tera,
     for_loops: Vec<ForLoop>,
+    should_escape: bool,
 }
 
 impl<'a> Renderer<'a> {
     pub fn new(tpl: &'a Template, tera: &'a Tera, context: Value) -> Renderer<'a> {
+        let mut should_escape = false;
+        for ext in &tera.autoescape_extensions {
+            if tpl.name.ends_with(ext) {
```

Comment:
`ends_with(ext)` will also match things like "file.not_html" or "this_is_my_html". Probably not intended. Should use something like `Path::new(&tpl.name).extension().map_or(false, |e| e == ext)`.